test/user.test.js

```diff
@@ -363,6 +363,88 @@ describe('User', function() {
     });
   });
 
+  describe('Password length validation', function() {
+    var pass72Char = new Array(73).join('a');
+    var pass73Char = new Array(74).join('a');
+    var passTooLong = pass72Char + 'WXYZ1234';
+
+    it('rejects passwords longer than 72 characters', function(done) {
+      try {
+        User.create({ email: 'b@c.com', password: pass73Char }, function(err) {
+          if (err) return done (err);
+          expect(err).to.have.property('message', /Password too long/);
+        });
+      } catch (e) {
+        done();
+      }
```

Comment:
As we tried together, nothing works except the `try/catch` .. Also, the only way this function is tested elsewhere is also using `try/catch`. If I had to remove that, I would have to dig more in `dao.js` and disect the function `Create`, which is not the goal of this pr because focus would be lost. The test case works and the proper error message is obtained. 

@bajtos @raymondfeng Any thoughts?